### DISCUSSION TITLE
Why is the bool 'asNum' method not 'toNum'?

### DISCUSSION THREAD

**Discussion by Calamity210** [SenegalLang/Senegal#12] [Help] [answered]
Asked by @danialgoodwin (#11):
> Hi, I was just wondering, what was the reasoning behind using `Bool.asNum()`, but `String.toNum()`? (source: docs)
> Sidenote: For my own language, I've been experimenting with only using `as*()` and no `to*()`

**Comment by Calamity210** [comment, marked as answer]
Its a great question, there are several differences between the two.

Firstly, it should be understood that the only false values in senegal are:
- null
- false
- 0

Anything else is considered to be true. That goes to say, any non-zero number is "true" while zero is "false".

Why does that matter? All numbers are valid within a boolean context, take an if clause for example:

```js
if (1)
 println("foo");
```
Is perfectly acceptable, you should expect foo to be printed. Thus, a numerical value can be used in place of a bool. 

This does not hold true for strings, a string cannot guarantee to be a numerical value, ie: `"foo123"`.  Even if a string was to assert that it contains nothing but numerical values, a character and num are not equivalent. The character '7' would be equivalent to the ascii code `55`.

To sum it up, a strings characters need to be *converted* to their expected numerical counterparts. Whereas senegal will interpret a number as a bool just fine.

**Reply by danialgoodwin** [reply]
Cool, sounds good to me :)
And, yeah, I definitely like the languages with the truthy/falsey if-conditions
